plugins/modules/ndo_l3out_interface_routing_policy.py

```diff
@@ -0,0 +1,856 @@
+                        ops.append(
+                            dict(
+                                op="replace",
+                                path="{0}/bfdPol/echoRxInterval".format(interface_routing_policy_path),
+                                value=bfd_settings.get("echo_receive_interval"),
+                            )
+                        )
+
+                    if bfd_settings.get("echo_admin_state") is not None and proposed_payload.get("bfdPol").get("echoAdminState") != bfd_settings.get(
+                        "echo_admin_state"
+                    ):
+                        proposed_payload["bfdPol"]["echoAdminState"] = bfd_settings.get("echo_admin_state")
+                        ops.append(
+                            dict(
+                                op="replace",
+                                path="{0}/bfdPol/echoAdminState".format(interface_routing_policy_path),
+                                value=bfd_settings.get("echo_admin_state"),
+                            )
+                        )
+
+                    interface_control_value = ENABLED_DISABLED_BOOLEAN_MAP.get(bfd_settings.get("interface_control"))
+                    if interface_control_value is not None and proposed_payload.get("bfdPol").get("gracefulRestartHelper") is not interface_control_value:
+                        proposed_payload["bfdPol"]["gracefulRestartHelper"] = interface_control_value
+                        ops.append(
+                            dict(
+                                op="replace",
+                                path="{0}/bfdPol/gracefulRestartHelper".format(interface_routing_policy_path),
+                                value=interface_control_value,
+                            )
+                        )
+
+            # OSPF Interface Settings
+            if ospf_interface_settings is not None:
+                if ospf_interface_settings.get("state") == "disabled" and proposed_payload.get("ospfIntfPol"):
+                    proposed_payload.pop("ospfIntfPol", None)
+                    ops.append(dict(op="remove", path="{0}/ospfIntfPol".format(interface_routing_policy_path)))
+
+                elif ospf_interface_settings.get("state") != "disabled":
+                    if not proposed_payload.get("ospfIntfPol"):
+                        proposed_payload["ospfIntfPol"] = dict()
+                        proposed_payload["ospfIntfPol"]["ifControl"] = dict()
+                        ops.append(dict(op="replace", path="{0}/ospfIntfPol".format(interface_routing_policy_path), value=dict()))
+                        ops.append(dict(op="replace", path="{0}/ospfIntfPol/ifControl".format(interface_routing_policy_path), value=dict()))
+
+                    network_type_value = get_ospf_network_type(ospf_interface_settings.get("network_type"))
+                    if network_type_value is not None and proposed_payload.get("ospfIntfPol").get("networkType") != network_type_value:
```

Comment:
I think all the OSPF value replace operations can be encapsulated into a function.
The option value along with the equivalent API option name can be inputs and then everything in the if statements can be a function.
The same can be done for BFD and BFD multihop dictionaries.

Reply:
Sharing an example function:
```
def set_replace_operations(ops, proposed_payload, interface_routing_policy_path, input_class, input_name, input_value):
    if input_value is not None and proposed_payload.get(input_class).get(input_name) != input_value:
        proposed_payload[input_class][input_name] = input_value
        ops.append(
            dict(
                op="replace",
                path="{0}/{1}/{2}".format(interface_routing_policy_path, input_class, input_name),
                value=input_value,
            )
        )
```
The arguments and function name can be adjusted of course. 

Then each option in the OSPF dictionary can be replaced with:
```
set_replace_operations(ops, proposed_payload, interface_routing_policy_path, "ospfIntfPol", "networkType", network_type_value)
set_replace_operations(ops, proposed_payload, interface_routing_policy_path, "ospfIntfPol", "prio", ospf_interface_settings.get("priority"))
```
A similar function or the same function can be used for the BFD options too.

Additionally to reduce code `ops`, `proposed_payload` and `interface_routing_policy_path` could be global to reduce the arguments required in the function. However, not sure if global variables are really recommended.

Reply:
Thank you for your feedback @samiib